tests/ui/issues/issue-61894.rs

```diff
@@ -17,5 +17,5 @@ impl<M> Bar<M> {
 }
 
 fn main() {
-    assert_eq!(Bar(()).foo(), "issue_61894::Bar<_>::foo::f");
+    assert_eq!(Bar(()).foo(), "issue_61894::Bar<M>::foo::f");
```

Comment:
What is confusing about this is that there is no `M` in scope when printing the name. If `M` shadows the name of some other type, this could become misleading.

Reply:
yea, iirc printing `_` was on purpose, because it can't have been relevant for the type name if it's still an uninstantiated param. If there are two `impl<M> Bar<M>` (e.g. with different bounds, which is where such "dupliate impl blocks" are common), then the type names would still be the same, so adding the `M` does not really bring any benefit